lib/transport.ts

```diff
@@ -0,0 +1,90 @@
+/*
+ * Unless explicitly stated otherwise all files in this repository are licensed
+ * under the Apache License Version 2.0.
+ *
+ * This product includes software developed at Datadog (https://www.datadoghq.com/).
+ * Copyright 2021 Datadog, Inc.
+ */
+
+import * as lambda from "@aws-cdk/aws-lambda";
+
+export const apiKeyEnvVar = "DD_API_KEY";
+export const apiKeyKMSEnvVar = "DD_KMS_API_KEY";
+export const siteURLEnvVar = "DD_SITE";
+export const logForwardingEnvVar = "DD_FLUSH_TO_LOG";
+
+export const transportDefaults = {
+  site: "datadoghq.com",
+  flushMetricsToLogs: true,
+  enableDDTracing: true,
+};
+
+export const site_list: string[] = [
+  "datadoghq.com",
+  "datadoghq.eu",
+  "us3.datadoghq.com",
+  "ddog-gov.com",
+];
+
+export class Transport {
+  flushMetricsToLogs: boolean;
+  site: string;
+  apiKey?: string;
+  apiKMSKey?: string;
+
+  constructor(
+    flushMetricsToLogs?: boolean,
+    site?: string,
+    apiKey?: string,
+    apiKMSKey?: string
+  ) {
+    if (flushMetricsToLogs === undefined) {
+      this.flushMetricsToLogs = transportDefaults.flushMetricsToLogs;
+    } else {
+      this.flushMetricsToLogs = flushMetricsToLogs;
+    }
+
+    if (site === undefined) {
+      this.site = transportDefaults.site;
+    } else {
+      this.site = site;
+    }
+
+    if (!site_list.includes(this.site.toLowerCase())) {
+      throw new Error(
```

Comment:
This means we can't test the cdk with out staging url. Could we just print a warning instead?

Reply:
Gotcha, fixed it by console.logging out the error message instead. 

```
if (!site_list.includes(this.site.toLowerCase())) {
      console.log("Warning: Invalid site URL. Must be either datadoghq.com, datadoghq.eu, us3.datadoghq.com, or ddog-gov.com.")
    }
```